### PR TITLE
Created request function to handle API calls

### PR DESCRIPTION
**Pull request by Nick-NCSU**
# Description

Created request function which attempts to fetch multiple times.

## Type of change

Please leave any irrelevant options unchecked.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [ ] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -31,6 +31,50 @@ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
 export class APIClient {
   constructor(readonly config: IntegrationConfig) {}
 
+  private readonly REQUEST_CONFIG = {
+    RETRY_METHODS: ['GET', 'PUT', 'DELETE'],
```
Comment (ndowmon) [inline src/client.ts:35]: Should we also retry `POST`? Seems like `RETRY_METHODS` is a proxy for idempotent methods, but we're just making LIST calls with our `POST` requests so they are idempotent.

**Review by Nick-NCSU — COMMENTED** [review]
Maybe adding an optional override for retry to make it clear we want to retry this request would be best?

File: src/client.ts [hunk 2]
```diff
@@ -31,6 +31,51 @@ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
...
+  private async request(
+    url: RequestInfo,
+    init?: RequestInit,
+    retry?: boolean,
+  ): Promise<Response> {
+    // If there is no init or init.method then this is a GET request.
+    retry ??=
+      !init ||
+      !init.method ||
+      this.REQUEST_CONFIG.RETRY_METHODS.includes(init.method);
```
Comment (Nick-NCSU) [inline src/client.ts:50]: Like so

**Review by ndowmon — APPROVED** [review]
Looks good, thanks for the quick turnaround. Posted a question about retrying `POST` but I don't think we're seeing `502`s on `POST` anyway, so definitely not blocking!

**Pull request merged**